### PR TITLE
Add `MemoryStore` to `--store-type` option

### LINKED ISSUE #1438: Drop `LiteDB` support and `--store-type` option
I think that we aren't using `LiteDB` implementation for 9c seriously.

### PR DESCRIPTION
**Pull request by U-lis**
resolves #1438 

---

### Description
Add `memory` type into `--store-type` option to use `MemoryStore` as blockchain data storage.

### Acknowledgement
- I did not add MemoryStore to `ChainCommandTest` type test because MemoryStore vanishes when it is disposed.
  - See comment in `ChainCommandTest.cs`

### CODE REVIEW

File: NineChronicles.Headless.Executable.Tests/Commands/ChainCommandTest.cs [hunk 1]
```diff
@@ -27,6 +26,10 @@ namespace NineChronicles.Headless.Executable.Tests.Commands
 {
     public class ChainCommandTest : IDisposable
     {
+        /// <summary>
+        /// `StoreType.Memory` will not be tested.
+        /// Because the purpose of ChainCommandTest is to store and read blockchain data.
+        /// </summary>
```
Comment (moreal) [inline NineChronicles.Headless.Executable.Tests/Commands/ChainCommandTest.cs:32]: In my thought, it shouldn't be an XML document here. The document will describe `ChainCommandTest._console` variable. How about describing it above the `ChainCommandTest` as normal documents (`//` not `///`)?

File: NineChronicles.Headless.Executable/Commands/ChainCommand.cs [hunk 2]
```diff
@@ -71,7 +71,7 @@ public void Help([FromService] ICoconaHelpMessageBuilder helpMessageBuilder)
                                "mimisbrunnr) of a given chain in csv format.")]
         public void Inspect(
             [Argument("STORE-TYPE",
-                Description = "Store type of RocksDb.")]
+                Description = "Store type of blockchain storage.")]
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Commands/ChainCommand.cs:74]: I don't have a detailed plan but I think it'll be better to restrict the `Memory` store type option here yet.

File: NineChronicles.Headless.Executable.Tests/Commands/ChainCommandTest.cs (hunk 1 above)
Comment (U-lis) [reply NineChronicles.Headless.Executable.Tests/Commands/ChainCommandTest.cs:32]: I didn't know the difference of `///` and `//` type comments.
I'll change them to normal comment.

File: NineChronicles.Headless.Executable/Commands/ChainCommand.cs (hunk 2 above)
Comment (U-lis) [reply NineChronicles.Headless.Executable/Commands/ChainCommand.cs:74]: So, you mean it'll be better to leave this description to unchanged?
I changed this because of the default option(`LiteDB`) can be used.
How about adding available options to description?

Comment (moreal) [reply NineChronicles.Headless.Executable/Commands/ChainCommand.cs:74]: Oops, sorry for confusing you with the selected lines. I mean that `MemoryStore` will be useless in the command because the `MemoryStore` is volatile, so I suggested restricting the type 🤔

File: NineChronicles.Headless.Executable/Commands/ChainCommand.cs [hunk 3]
```diff
@@ -127,7 +142,7 @@ public void Help([FromService] ICoconaHelpMessageBuilder helpMessageBuilder)
 
             var typeOfActionTypeAttribute = typeof(ActionTypeAttribute);
             foreach (var item in
-                store.IterateIndexes(chain.Id, offset + 1 ?? 1, limit).Select((value, i) => new { i, value }))
+                     store.IterateIndexes(chain.Id, offset + 1 ?? 1, limit).Select((value, i) => new { i, value }))
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Commands/ChainCommand.cs:145]: Is this intended?

File: NineChronicles.Headless.Executable/Commands/ChainCommand.cs [hunk 4]
```diff
@@ -61,17 +68,19 @@ public void Help([FromService] ICoconaHelpMessageBuilder helpMessageBuilder)
             }
 
             BlockHash tipHash = store.IndexBlockHash(chainId, -1)
-                          ?? throw new CommandExitedException("The given chain seems empty.", -1);
+                                ?? throw new CommandExitedException("The given chain seems empty.", -1);
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Commands/ChainCommand.cs:71]: Is this intended? it seems not related with this pull request 🤔

File: NineChronicles.Headless.Executable/Commands/ChainCommand.cs [hunk 5]
```diff
@@ -243,7 +265,8 @@ public void Help([FromService] ICoconaHelpMessageBuilder helpMessageBuilder)
                 }
 
                 snapshotTipHash = hash;
-            } while (!stateStore.ContainsStateRoot(store.GetBlock<NCAction>(hashAlgorithmGetter, snapshotTipHash).StateRootHash));
+            } while (!stateStore.ContainsStateRoot(store.GetBlock<NCAction>(hashAlgorithmGetter, snapshotTipHash)
+                         .StateRootHash));
```
Comment (moreal) [inline NineChronicles.Headless.Executable/Commands/ChainCommand.cs:269]: Is this intended?

**Review by moreal — COMMENTED** [review]
Thanks for the nice work! I left some comments about style, as the last review! 🙏🏻

**Comment by U-lis** [comment]
@moreal  I didn't indented your comments. Maybe It's Rider's auto reformatting.
I'll double check those type of things. Sorry about messy commit history and I'll fix and push right away.

**Review by moreal — APPROVED** [review]
Thank you for your contribution!
